examples/hitl/rearrange_v2/ui.py

```diff
@@ -633,6 +661,204 @@ def _update_hovered_object_ui(self):
 
     def _update_selected_object_ui(self):
         """Draw a UI for the currently selected object."""
+        object_id = self._click_selection.object_id
+
+        object_category_name: Optional[str] = None
+        object_state_controls: List[ObjectStateControl] = []
+        primary_region_name: Optional[str] = None
+        contextual_info: Optional[str] = None
+        contextual_color: Optional[List[float]] = None
+
+        if object_id is None:
+            return
+
+        if object_id is not None:
+            world = self._world
+            sim = self._sim
+            obj = sutils.get_obj_from_id(
+                sim, object_id, world._link_id_to_ao_map
+            )
+            if obj is not None:
+                # Get object category name.
+                object_category_name = world.get_category_from_handle(
+                    obj.handle
+                )
+                if object_category_name is None:
+                    object_category_name = "Object"
+
+                # Get object state controls.
+                object_state_controls = self._get_object_state_controls(
+                    obj.handle
+                )
+
+                # Get the primary region name.
+                primary_region = world.get_primary_object_region(obj)
+                if primary_region is not None:
+                    primary_region_name = primary_region.category.name()
+
+                # Get the contextual information.
```

Comment:
This section will be refactored in a later pass.
This will unify constraints across the application and clean up the UI code.

@aclegg3 : I'll wait until your constraint changes are applied before doing this.